src/main/java/com/iexec/worker/result/ResultService.java

```diff
@@ -302,11 +300,9 @@ public String getIexecUploadToken() {
         // get challenge
         Integer chainId = blockchainAdapterConfigurationService.getChainId();
         try {
-            Eip712Challenge eip712Challenge = resultProxyClient.getChallenge(chainId);
+            EIP712Challenge eip712Challenge = resultProxyClient.getChallenge(chainId);
             // sign challenge
-            ECKeyPair ecKeyPair = credentialsService.getCredentials().getEcKeyPair();
-            String signedEip712Challenge = Eip712ChallengeUtils.buildAuthorizationToken(eip712Challenge,
-                    workerConfigService.getWorkerWalletAddress(), ecKeyPair);
+            String signedEip712Challenge = eip712Challenge.buildAuthorizationTokenFromCredentials(credentialsService.getCredentials());
```

Comment:
I would be safer to:
1. Verify domain before signing
Something like:
`if(eip712Challenge.getName() == "iExec Core" && eip712Challenge.getChainId() == 17){ ...`
2. Keep the keys inside the `credentialsService`
Something like:
`credentialsService.signEip712Challenge(eip712Challenge)` plus a method building the auth token (like you do)
Or for no-duplication purposes:
`credentialsService.signEip712ChallengeAndBuildToken(eip712Challenge)`

Reply:
Thanks!